Styled text and search fields override their backgrounds with alternate form control design enabled

hortont424 opened the pull request:
#### 2efdbd54bbada1a21df36fd13f6031a35712300b
<pre>
Styled text and search fields override their backgrounds with alternate form control design enabled
<a href="https://bugs.webkit.org/show_bug.cgi?id=253093">https://bugs.webkit.org/show_bug.cgi?id=253093</a>
rdar://105803404

Reviewed by Aditya Keerthi.

Web content that overrides certain CSS properties on input elements is specified
to disable the UA appearance in <a href="https://drafts.csswg.org/css-ui/#appearance-disabling-properties">https://drafts.csswg.org/css-ui/#appearance-disabling-properties</a>

Our existing code only covers a subset of these properties, and only on a subset
of text-field-like input fields.

* Source/WebCore/rendering/RenderThemeIOS.mm:
(WebCore::RenderThemeIOS::isControlStyled const):
FillLayer does not include color, so overriding background color did not drop you out of the UA appearance.
Similarly, adjustments to borders are supposed to drop you out of UA appearance.
Fix both things by adopting borderAndBackgroundEqual, which tests the FillLayers, background color, and borders.

Add search fields to the set of text-field-like fields that get this behavior, as well.

Canonical link: <a href="https://commits.webkit.org/261073@main">https://commits.webkit.org/261073@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/260a5a936822bbed2b345caf8449ac4619f64473

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/110460 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/19547 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/43051 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/87/builds/1834 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/119363 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/114407 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/20978 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/10695 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/102698 "Built successfully") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/116202 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/15624 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/98809 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/43850 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/97585 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/30475 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/85717 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/12203 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/31811 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/12781 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/8780 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/18144 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/51430 "Passed tests") | | 
| [❌ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/7672 "Failed to push commit to Webkit repository") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/14645 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

pxlcoder reviewed (approved):
Can you add a link to the spec in the commit message?

https://drafts.csswg.org/css-ui/#appearance-disabling-properties

Also, we should add a test if there isn't a new WPT pass.

hortont424 committed:
Styled text and search fields override their backgrounds with alternate form control design enabled
https://bugs.webkit.org/show_bug.cgi?id=253093
rdar://105803404

Reviewed by Aditya Keerthi.

Web content that overrides certain CSS properties on input elements is specified
to disable the UA appearance in https://drafts.csswg.org/css-ui/#appearance-disabling-properties

Our existing code only covers a subset of these properties, and only on a subset
of text-field-like input fields.

* Source/WebCore/rendering/RenderThemeIOS.mm:
(WebCore::RenderThemeIOS::isControlStyled const):
FillLayer does not include color, so overriding background color did not drop you out of the UA appearance.
Similarly, adjustments to borders are supposed to drop you out of UA appearance.
Fix both things by adopting borderAndBackgroundEqual, which tests the FillLayers, background color, and borders.

Add search fields to the set of text-field-like fields that get this behavior, as well.

Canonical link: https://commits.webkit.org/261073@main
